lib/web3/syncing.js

```diff
@@ -63,23 +59,18 @@ var pollSyncing = function(self) {
         });
     };
```

Comment:
I reviewed this class carefully now and it is not well designed, nor reliable. I would not include it in `web3.js` and/or use it in production code.

not reliable / not stateless
```js
var cb1 = function (e, result) {};
var cb2 = function (e, result) {};
var s1 = web3.eth.isSyncing(cb1);

// poll1 returns false, cb1 is not called
// poll2 returns false, cb1 is not called
// poll3 returns {sb: 0x1, cb: 0x2, hb: 0x3} , cb1 is called twice:
//  cb1(null, true)
//  cb1(null,  {sb: 0x1, cb: 0x2, hb: 0x3})
// poll4 returns {sb: 0x1, cb: 0x2, hb: 0x3} , cb1 is called twice:
//  cb1(null, true)
//  cb1(null,  {sb: 0x1, cb: 0x2, hb: 0x3})

var s2 = web3.eth.isSyncing(cb2);
// poll5 returns false, cb1 is called, cb2 is not called
//  cb1(null, false)
```

There are visible two inconsistencies:
- if first poll returns false, callback will not be triggered
- if current poll returns false, and previous poll also returned false, callback will not be triggered. Whereas if current poll returned the same object as previous one, it will trigger callback.

Design issues:
- `addCallback` method is redundant and introduces unnecessary complexity. There is already `web3.eth.syncing` method which does the same thing.

There are also few other small issues:

- prototype constructor `returns this`. It's redundant
- It's possible to create 2 `IsSyncing` objects with the same `this.pollId`. Instead of using random number. You should use static number and increase it by 1 every time you create new instance of the object.

Tips:
- `setTimeout(function() {}, 1)` means almost the same as `setTimeout(function() {}, 0)` and `setTimeout(function() {})`. In all cases function will be processed asynchronously

Suggestions:
- remove `lastSyncState`. Always return what api returns.
- remove `callback(null, true)`. Imo it's not necessary. Callback function must check typeof `result` anyway
    
```js
  // call the callback with true first so the app can stop anything, before receiving the sync data
  if(!lastSyncState && utils.isObject(sync))
    callback(null, true);
```

- remove `this.callbacks` and `addCallback`. It will simplify everything.

I would also insist on renaming this function. `IsSyncing` implies that it returns `true`, `false` synchronously.

Please consider these changes and apply them in separate pr. For this pr, just please just fix `IsSyncing` tests and/or revert the changes in `IsSyncing` prototype.